snippets/product-media-gallery.liquid

```diff
@@ -154,6 +156,13 @@
         id="Slider-Thumbnails-{{ section.id }}{{ id_append }}"
         class="thumbnail-list list-unstyled slider slider--mobile{% if section.settings.gallery_layout == 'thumbnail_slider' %} slider--tablet-up{% endif %}"
       >
+        {%- capture sizes -%}
+          (min-width: {{ settings.page_width }}px) calc(({{ settings.page_width | minus: 100 | times: media_width | round }} - 4rem) / 4),
+          (min-width: 990px) calc(({{ media_width | times: 100 }}vw - 4rem) / 4),
+          (min-width: 750px) calc((100vw - 15rem) / 8),
+          calc((100vw - 8rem) / 3)
```

Comment:
I will say the crux of this PR is the image size stuff and there could be cases I'm missing, but like I mentioned about the srcset, with the cover fit in place, I've kind of been rounding up to avoid having to create an unnecessarily complex sets of conditionals to account for possible setting/screen combination.

For this one specifically, I know there was a specific mobile case where a landscape image was rendering very blurry with the original size. I believe I sort of combined the existing mobile size logic here with the "other" (un-updated) instance in the loop below which was originally `calc((100vw - 8rem) / 5)`. The 5 seemed like a mistake because mobile only shows 3 thumbs, but I otherwise felt this had a better result than `calc((100vw - 14rem) / 3)`

I don't know, maybe something like 11rem would be better than 8 but I'm not sure where 14 came from regardless.
<img src="https://screenshot.click/26-53-0dr1g-pl9ot.png" width="200" />